alibi_detect/cd/base.py

```diff
@@ -0,0 +1,215 @@
+from abc import abstractmethod
+from functools import partial
+import logging
+import numpy as np
+from typing import Callable, Dict, Optional, Tuple, Union
+from alibi_detect.base import BaseDetector, concept_drift_dict
+from alibi_detect.cd.preprocess import preprocess_drift
+from alibi_detect.cd.utils import update_reference
+from alibi_detect.utils.statstest import fdr
+
+logger = logging.getLogger(__name__)
+
+
+class BaseUnivariateDrift(BaseDetector):
+
+    def __init__(self,
+                 p_val: float = .05,
+                 X_ref: Union[np.ndarray, list] = None,
```

Comment:
Can this actually be a `list` (same with the child classes)? If not, suggest changing to `np.ndarray` and then we can remove some of the `type: ignore` directives when getting the shape of `X_ref` below.

Reply:
Yes, for text.

Reply:
Is there something we can add to the docstrings about this? Is it still needed for e.g. Tabular? If no, we should remove it from the signature, if yes, we should check with `isinstance` and do the right thing which would also get rid of the `# type: ignore` directives when calling `shape`.